lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -33,11 +33,14 @@ def _host_pool_current_database=(database)
     end
 
     def self.ruby2_keywords(*); end unless respond_to?(:ruby2_keywords, true)
-    ruby2_keywords def execute_with_switching(*args, **kwargs)
+    # This one really does need ruby2_keywords; in Rails 6.0 the method does not take
+    # any keyword arguments, but in Rails 7.0 it does. So, we don't know whether or not
+    # what we're delegating to takes kwargs, so ruby2_keywords is needed.
+    ruby2_keywords def execute_with_switching(*args)
       if _host_pool_current_database && !_no_switch
         _switch_connection
       end
-      execute_without_switching(*args, **kwargs)
+      execute_without_switching(*args)
```

Comment:
won't this drop the keyword arguments in rails 7?

does it need to be defined differently for rails 7 i.e.

```
if ActiveRecord::VERSION::MAJOR >= 7
  # rails 7 isn't supposed to work with ruby 2.6 anyway
  def execute_with_switching(*args, **kwargs)
    ...
    execute_without_switching(*args, **kwargs)
  end
else
  ruby2_keywords def execute_with_switching(*args)
    ...
    execute_without_switching(*args)
  end
end




```

Reply:
oh nevermind i think i get it now